format/composite.go

```diff
@@ -345,6 +263,84 @@ func (c *CompositeFormatter) Close(ctx context.Context) error {
 	return nil
 }
 
+// Hash takes anything that might affect the paths to be traversed, or how they are traversed, and adds it to a sha256
+// hash.
+// This can be used to determine if there has been a material change in config or setup that requires the cache
+// to be invalidated.
+func (c *CompositeFormatter) Hash() (string, error) {
+	h := sha256.New()
+
+	// start with the global excludes
+	h.Write([]byte(strings.Join(c.cfg.Excludes, " ")))
+	h.Write([]byte(strings.Join(c.cfg.Global.Excludes, " ")))
```

Comment:
I don't understand why the excludes matter for caching. I'd expect those to only affect filesystem walking, which IIUC, is not something we cache.

For example:

1. My project has 2 files: `a.py` and `b.py`. `b.py` is currently excluded.
2. I format my whole project. I get a cache entry for `a.py`.
3. I then change my excludes so `b.py` is now included.
4. I format my whole project. I see a cache entry for `a.py`, so I have nothing to do there. I don't see a cache entry for `b.py`, so I format it and add an entry.

Reply:
You make a fair point. We don't need to include excludes, both globally and at the formatter level, in the hash.